crates/bevy_render/src/world_sync.rs

```diff
@@ -0,0 +1,261 @@
+use bevy_app::Plugin;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    observer::Trigger,
+    query::With,
+    reflect::ReflectComponent,
+    system::{Local, Query, ResMut, Resource, SystemState},
+    world::{Mut, OnAdd, OnRemove, World},
+};
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_reflect::Reflect;
+use bevy_utils::tracing::warn;
+
+/// A Plugin that synchronizes entities with specific Components between the main world and the render world.
+///
+/// Bevy's renderer is architected independently from the main app.
+/// It operates in its own separate ECS World, so the renderer could run in parallel with the main world logic.
+/// This is called "Pipelined Rendering", see [`PipelinedRenderingPlugin`] for more information.
+///
+/// [`WorldSyncPlugin`] is the first thing that runs every frame and it maintains an entity-to-entity mapping
+/// between the main world and the render world.
```

Comment:
```suggestion
/// between the main world and the render world, by spawning new entities (without any components) in the render world to match newly spawned entities in the main world, and despawning entities in the render world to match newly despawned entities in the main world.
```

Could also omit these changes as I see that you mention it more later on in the docs, up to you. But I would like to keep the specific mention that when we spawn new entities, they do not have any components.